### PR TITLE
Remove IHttpClient from Providers

### PR DESCRIPTION
**Pull request by crobibero**

### CODE REVIEW

File: Jellyfin.Api/Controllers/ItemLookupController.cs [hunk 1]
```diff
@@ -331,12 +331,12 @@ public async Task<ActionResult<IEnumerable<RemoteSearchResult>>> GetBookRemoteSe
         /// <returns>Task.</returns>
         private async Task DownloadImage(string providerName, string url, Guid urlHash, string pointerCachePath)
         {
-            var result = await _providerManager.GetSearchImage(providerName, url, CancellationToken.None).ConfigureAwait(false);
-            var ext = result.ContentType.Split('/').Last();
+            using var result = await _providerManager.GetSearchImage(providerName, url, CancellationToken.None).ConfigureAwait(false);
+            var ext = result.Content.Headers.ContentType.MediaType.Split('/').LastOrDefault();
```
Comment (Bond-009) [inline Jellyfin.Api/Controllers/ItemLookupController.cs:335]: Why change `Last` to `LastOrDefault`?

File: Jellyfin.Api/Controllers/LiveTvController.cs [hunk 2]
```diff
@@ -1071,7 +1071,7 @@ public async Task<ActionResult> GetSchedulesDirectCountries()
         {
             var client = _httpClientFactory.CreateClient();
             // https://json.schedulesdirect.org/20141201/available/countries
-            using var response = await client.GetAsync("https://json.schedulesdirect.org/20141201/available/countries")
+            var response = await client.GetAsync("https://json.schedulesdirect.org/20141201/available/countries")
```
Comment (Bond-009) [inline Jellyfin.Api/Controllers/LiveTvController.cs:1074]: Why?

File: Jellyfin.Api/Helpers/FileStreamResponseHelpers.cs [hunk 3]
```diff
@@ -36,7 +36,7 @@ public static class FileStreamResponseHelpers
                 httpClient.DefaultRequestHeaders.Add(HeaderNames.UserAgent, useragent);
             }
 
-            using var response = await httpClient.GetAsync(state.MediaPath).ConfigureAwait(false);
+            var response = await httpClient.GetAsync(state.MediaPath).ConfigureAwait(false);
```
Comment (Bond-009) [inline Jellyfin.Api/Helpers/FileStreamResponseHelpers.cs:39]: Why?

File: Jellyfin.Api/Controllers/LiveTvController.cs (hunk 2 above)
Comment (crobibero) [reply Jellyfin.Api/Controllers/LiveTvController.cs:1074]: having `using` there disposes the stream before the aspnet framework can return it, resulting in `Cannot access a closed Stream.`

Comment (Bond-009) [reply Jellyfin.Api/Controllers/LiveTvController.cs:1074]: Maybe add a comment?

File: Jellyfin.Api/Controllers/ItemLookupController.cs [hunk 4]
```diff
@@ -331,12 +331,12 @@ public async Task<ActionResult<IEnumerable<RemoteSearchResult>>> GetBookRemoteSe
         /// <returns>Task.</returns>
         private async Task DownloadImage(string providerName, string url, Guid urlHash, string pointerCachePath)
         {
-            var result = await _providerManager.GetSearchImage(providerName, url, CancellationToken.None).ConfigureAwait(false);
-            var ext = result.ContentType.Split('/').Last();
+            using var result = await _providerManager.GetSearchImage(providerName, url, CancellationToken.None).ConfigureAwait(false);
+            var ext = result.Content.Headers.ContentType.MediaType.Split('/').Last();
```
Comment (Bond-009) [inline Jellyfin.Api/Controllers/ItemLookupController.cs:335]:
```suggestion
            var ext = result.Content.Headers.ContentType.MediaType.Split('/')[^1];
```
PERFORMANCE ;)

**Pull request merged**